app/src/main/java/nl/tudelft/b_b_w/view/FriendsPageActivity.java

```diff
@@ -15,39 +25,127 @@
 public class FriendsPageActivity extends Activity {
 
     /**
-     * Used to create connection with database
+     * block controller
      */
-    private GetDatabaseHandler getDatabaseHandler;
+    private BlockController bc;
 
     /**
-     * block controller
+     * Block argument to create a block
+     */
+    private static final String BLOCK = "BLOCK";
+
+    /**
+     * This is your own user
+     */
+    User user;
+
+    /**
+     * this is the paired contact
      */
-    private BlockController blockController;
+    User contact;
 
     /**
-     * the owner of the block
+     * the contact name
      */
-    private String ownerName;
+    private String contactName;
 
     /**
-     * the IBAN nummer of the owner
+     * the IBAN nummer of the contact
      */
     private String ibanNumber;
 
     /**
-     * the public key of the block
+     * the public key of the contact
      */
     private String publicKey;
 
     /**
+     * Block hash of the paired contact's genesis
+     */
+    private String contactGenesisBlockHash;
+
+    /**
+     * the hash of your latest block
+     */
+    private String userLatestBlockHash;
+
+    /**
+     * Text view of the iban number.
+     */
+    private TextView textViewIban;
+
+    /**
+     * Text view of the contact name.
+     */
+    private TextView textViewContact;
+
+
+    /**
      * On create method, here we request a database connection
      * @param savedInstanceState
      */
     @Override
     protected final void onCreate(Bundle savedInstanceState) {
+        bc = new BlockController(this);
+
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_friends_page);
 
+        SharedPreferences settings = getSharedPreferences(PREFS_NAME, 0);
+        contact = new User(settings.getString("userNameTestSubject", ""), settings.getString("ibanTestSubject", ""));
+        user = new User(settings.getString("userName", ""), settings.getString("iban", ""));
+
+        contactName = contact.getName();
```

Comment:
Oh no this was so because we used these value many time so I want to make it more oversee able.